proposals/lip-0050.md

```diff
@@ -304,37 +400,53 @@ registerKeysParamsSchema = {
 ```python
 def verify(trs: Transaction) -> None:
 
-    validatorAddress = 20-byte address derived from trs.senderPublicKey
+    validatorAddress = SHA-256(trs.senderPublicKey)[:NUM_BYTES_ADDRESS]
 
-    if validator.getValidatorAccount(validatorAddress) does not exists :
+    if Validators.getValidatorAccount(validatorAddress) is empty:
+        address = SHA-256(trs.senderPublicKey)[:NUM_BYTES_ADDRESS]
+        emitPersistentEvent(
+            module = MODULE_NAME_LEGACY,
+            name = EVENT_NAME_KEY_REGISTRATION_FAILED,
+            data = {
+                "address": validatorAddress,
+                "reason": KEY_REGISTRATION_FAILED_NO_VALIDATOR_ACCOUNT
+            },
+            topics = [validatorAddress]
+        )
```

Comment:
Same as for reclaim, no emitting in the verify function.

Reply:
Indeed, removed events from verification.